provider/core_dhcp_options_resource.go

```diff
@@ -139,6 +147,14 @@ func (s *DHCPOptionsResourceCrud) State() string {
 }
 
 func (s *DHCPOptionsResourceCrud) Create() (e error) {
+	// If we are creating a default resource, then don't have to
+	// actually create it. Just set the ID and update it.
+	if defaultId, ok := s.D.GetOk("manage_default_resource_id"); ok {
```

Comment:
I dot see a validation that makes sure either (compartmentID, vcnId ) or defaultId is present.

Reply:
If both aren't present, the service API will fail with a message that the comparmentId/vcnId isn't valid.

That being said, I will add some validation logic here to catch this earlier.